components/compiler/exprparser.cpp

```diff
@@ -324,6 +326,22 @@ namespace Compiler
                 mExplicit = name2;
                 return true;
             }
+
+            // In case someone tried passing an int in quotes
+            if (regex_match(name, std::regex("[+-]?[0-9]+")))
+            {
+                pushIntegerLiteral (stof(name));
```

Comment:
pushIntegerLiteral accepts an integer, not a floating point number, you should use stoi directly.